jbmc/src/java_bytecode/java_types.h

```diff
@@ -242,6 +242,32 @@ inline bool can_cast_type<java_class_typet>(const typet &type)
   return can_cast_type<class_typet>(type);
 }
 
+class java_method_typet : public code_typet
+{
+public:
+  const std::vector<irept> &throws_exceptions() const
+  {
+    return find(ID_exceptions_thrown_list).get_sub();
+  }
+
+  std::vector<irept> &throws_exceptions()
+  {
+    return add(ID_exceptions_thrown_list).get_sub();
+  }
+};
+
+inline const java_method_typet &to_java_method_type(const typet &type)
+{
+  PRECONDITION(type.id() == ID_code);
```

Comment:
To add a bit more type safety, you might want to add a `ID_C_java_method_type` so a caller can be sure that it really is a `java_method_typet` not a `code_typet`.

Reply:
Why would this be a comment with `C_` in the name instead of `ID_java_method_type`?

Reply:
Consistency with other types, which isn't a great reason. I think you're probably right in fact, if I have a C style function pointer that probably shouldn't be treated equivalently to a static java function that doesn't throw any exceptions.

Reply:
This has been added in https://github.com/diffblue/cbmc/pull/2661